### ISSUE TITLE
Large projects - long save time after auto-save (project folder)

### ISSUE THREAD

**Issue by Ldwebster111** [Scirra/Construct-3-bugs#5825]
## Problem description

When working on larger projects using the project folder, there are abnormally long save times after each autosave. After testing, this seems to be exponentially related to the size of the project being saved, with longer saves times seemingly related to the size of the project. The below attached project will take a couple of minutes to save, while my personal project (160mb) takes roughly 7 minutes after each save. This has been mentioned by other users across both Discord and the forum: https://www.construct.net/en/forum/construct-3/general-discussion-7/why-project-takes-long-save-167781

## Attach a .c3p

[slow saves.zip](https://github.com/Scirra/Construct-3-bugs/files/8996283/slow.saves.zip)

NOTE - the zip file is a project folder, not a .c3p. I've attempted to simulate a large project by adding multiple large layouts and sprites with a lot of frames.

## Steps to reproduce

1. Open up attached project folder, press save and note the quick time for the project to manually save 
2. Wait for Construct 3 to autosave (I'd suggest lowering the time in settings) ***
3. Press save again and note the much longer time it takes for the manual save to occur

*** Something I discovered while setting this up. Construct 3 won't autosave until the user has gone into the event sheet view at least once. As such, user may need to navigate across tabs a bit to trigger the autosave feature before moving to step 3.

## Observed result

Manual saves after autosaves take a long time. This becomes more apparent the larger a project becomes.

## Expected result

Manual saves should always be quick, especially as that is one of the main benefits for using project folders.

## More details

Disabling autosave removed this problem. Obviously, this isn't ideal, but becomes necessary once a project reaches a certain size as it otherwise becomes too much of a disruption to the work flow (my project was set to autosave every 10 minutes, which, as it was taking 7 minutes for a manual save, meant the vast majority of my time was spent looking at a saving dialogue).

Being a write time issue, I have tried this on three separate hard drives - a SATA, a generic SSD and an NVMe M.2 drive, all of which experienced this issue. Adding the faster drive to my system made no notable difference to save time.

**Affected browsers/platforms:**

Checked with Chrome and Edge standalone installs, Windows, stable.

**First affected release:**

Unsure on this. I know this has been happening for at least 6 months as I stopped using project folders around then as I was sick of waiting for (what I thought of at the time were) randomly long save times. I only switched back recently due to a separate issue with saving in .c3p files that made the project folder a better option for me.

## System details

tested on:

Win 10, Ryzen 5 3600, 16GB RAM, 6700XT

Win 10, Intel i5-4670k, 16GB RAM, 970GTX

**Comment by Ldwebster111** [comment]
From the above linked forum thread, dop2000 noted

"Auto-save feature saves the project as a single c3p file. Maybe Construct needs to rebuild the file structure or something to save as a folder after that. You can test this by manually saving as c3p and then as a folder. "

On testing this with the attached project file, I can confirm that this also triggers a slow project save.

**Comment by AshleyScirra** [comment]
Thanks for the report - it turns out autosave was accidentally marking all project images as changed (~8500 files in this case), causing them to be written back to the folder next time you saved. Writing 8500 files to disk should only take a few seconds, but unfortunately Chrome on Windows appears to invoke Windows Defender to scan each file written to disk one at a time, which is incredibly slow. (I reported this underling issue to Google some time ago: https://bugs.chromium.org/p/chromium/issues/detail?id=1255606)

I believe I've fixed this by making sure autosave does not mark images changed for folder saves. However we're close to a stable release and given that the saving code is high stakes, the fix will be held back until the next release cycle (it will be in the first beta after the next stable release), to maximise the testing time.
